test-data/unit/check-flags.test

```diff
@@ -410,6 +410,44 @@ async def h() -> NoReturn:  # E: Implicit return in function which does not retu
 [builtins fixtures/dict.pyi]
 [typing fixtures/typing-async.pyi]
 
+[case testNoWarnNoReturn]
+# flags: --no-warn-no-return --strict-optional
+import typing
+
+def implicit_optional_return(arg) -> typing.Optional[str]:
```

Comment:
What will happen for a case like:

```python
def implicit_optional_no_return() -> Optional[int]:
    print(1)  # or any other non-primitive body
```

Reply:
And what will happen for just `-> int`?

Reply:
No error for the first, error for the second `Incompatible return value type (implicitly returns "None", expected "int")`. Same as the existing test cases, since the `if` statement has no impact on reachability.